apollo-api/src/main/java/com/spotify/apollo/Request.java

```diff
@@ -134,6 +142,24 @@
   Request withPayload(ByteString payload);
 
   /**
+   * Creates a new {@link Request} based on this, but with a different ttl.
+   *
+   * @param ttl The duration, will be turned into ms
+   */
+  default Request withTtl(final Duration ttl) {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Creates a new {@link Request} based on this, but with a different ttl in ms
+   *
+   * @param ttl The duration in ms
+   */
+  default Request withTtl(final int ttl) {
```

Comment:
I don't think we should have this - it's better to have as few methods as possible, and in my opinion, this doesn't carry its weight.